static/js/components/CentroidPlot.jsx

```diff
@@ -0,0 +1,328 @@
+import React, { useEffect } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import PropTypes from "prop-types";
+import embed from "vega-embed";
+import * as d3 from "d3";
+import * as photometryActions from "../ducks/photometry";
+
+// Helper functions for computing plot points (taken from GROWTH marshall)
+const gcirc = (ra1, dec1, ra2, dec2) => {
+  // input deg, haversine formula, return deg
+  ra1 = (ra1 / 180) * Math.PI;
+  dec1 = (dec1 / 180) * Math.PI;
+  ra2 = (ra2 / 180) * Math.PI;
+  dec2 = (dec2 / 180) * Math.PI;
+  const delDec2 = (dec2 - dec1) * 0.5;
+  const delRA2 = (ra2 - ra1) * 0.5;
+  return (
+    (360 *
+      Math.asin(
+        Math.sqrt(
+          Math.sin(delDec2) * Math.sin(delDec2) +
+            Math.cos(dec1) *
+              Math.cos(dec2) *
+              Math.sin(delRA2) *
+              Math.sin(delRA2)
+        )
+      )) /
+    Math.PI
+  );
+};
+
+const relativeCoord = (ra, dec, refRA, refDec) => {
+  const delRA = gcirc(ra, dec, refRA, dec) * 3600 * -Math.sign(ra - refRA);
+  const delDec = gcirc(ra, dec, ra, refDec) * 3600 * Math.sign(dec - refDec);
+  return { delRA, delDec };
+};
+
+// Temporary function for draft
+const getReferencePoint = (ras, decs) => {
+  const refRA = d3.median(ras);
+  const refDec = d3.median(decs);
+
+  return { refRA, refDec };
+};
+
+const getCirclePoints = (delRaGroup, delDecGroup) => {
+  const thetas = d3.range(0, 2.01 * Math.PI, 0.01);
+  const C = Math.max(d3.deviation(delRaGroup), d3.deviation(delDecGroup));
+  const medianRA = d3.median(delRaGroup);
+  const medianDec = d3.median(delDecGroup);
+
+  const points = thetas.map((theta) => {
+    const xx = medianRA + C * Math.cos(theta);
+    const yy = medianDec + C * Math.sin(theta);
+    return { xx, yy, theta };
+  });
+
+  return points;
+};
+
+const getMessages = (delRaGroup, delDecGroup) => {
+  const offset = Math.sqrt(
+    d3.median(delRaGroup) ** 2 + d3.median(delDecGroup) ** 2
+  );
+  const C = Math.max(d3.deviation(delRaGroup), d3.deviation(delDecGroup));
+
+  const offsetMessage = {
+    x: 0.6,
+    y: 0.8,
+    message: `offset = ${offset.toFixed(2)} \u00B1 ${C.toFixed(2)}`,
+  };
+
+  return [offsetMessage];
+};
+
+// The Vega-Lite specifications for the centroid plot
+const spec = (inputData) => ({
+  $schema: "https://vega.github.io/schema/vega-lite/v4.json",
+  width: 400,
```

Comment:
Yep, makes sense. I just hard-coded it this way because I saw the other plots on the Source page were. It shouldn't be hard to make it responsive later.